### PR TITLE
(gcr) more reliable way of retrieving service account key

### PR DESCRIPTION
**Pull request by lwander**
@ewiseblatt @jtk54

### CODE REVIEW

File: install/first_google_boot.sh [hunk 1]
```diff
@@ -267,9 +267,9 @@ function extract_spinnaker_gcr_credentials() {
     echo "GCR enabled"
 
     if [ -z "$gcr_account" ]; then
-      # the @developer address is reserved for the project's special service
-      # compute engine default service account.
-      gcr_account=$(gcloud iam service-accounts list --format='value(email)' | grep developer -m 1)
+      # This service account is enabled with the Compute API.
+      project_id=$(gcloud projects list --format="value(projectNumber)" --filter="projectId:$MY_PROJECT")
```
Comment (ewiseblatt) [inline install/first_google_boot.sh]: Is project_id the number?
If so, you can get this from the metadata service directly.

Comment (lwander) [inline install/first_google_boot.sh]: Oh that's much better - I'll do that - thanks

Comment (lwander) [inline install/first_google_boot.sh]: Done

File: install/first_google_boot.sh [hunk 2]
```diff
@@ -325,6 +324,7 @@ MY_ZONE=""
 if full_zone=$(curl -s -H "Metadata-Flavor: Google" "$INSTANCE_METADATA_URL/zone"); then
   MY_ZONE=$(basename $full_zone)
   MY_PROJECT=$(curl -s -H "Metadata-Flavor: Google" "$METADATA_URL/project/project-id")
+  MY_PROJECT_NUMBER=$(curl -s -H "Metadata-Flavor: Google" "$METADATA_URL/project/numeric-project-id")
```
Comment (ewiseblatt) [inline install/first_google_boot.sh]: get_instance_metadata_attribute(

Comment (lwander) [inline install/first_google_boot.sh]: What do you mean?

Comment (ewiseblatt) [inline install/first_google_boot.sh]: I was looking at something and bus hit a bump causing me to add a comment I didnt mean to.

Comment (ewiseblatt) [inline install/first_google_boot.sh]: LGTM

Comment (lwander) [inline install/first_google_boot.sh]: Haha - ok thanks

**Pull request merged**